Source/IGListCollectionViewLayout.mm

```diff
@@ -577,19 +590,23 @@ - (void)calculateLayoutIfNeeded {
                 rollingSectionBounds = CGRectUnion(rollingSectionBounds, frame);
             }
         }
-
+       
         const CGRect headerBounds = (self.scrollDirection == UICollectionViewScrollDirectionVertical) ?
                 CGRectMake(insets.left,
-                        CGRectGetMinY(rollingSectionBounds) - headerSize.height,
+                        (itemCount == 0) ? CGRectGetMaxY(rollingSectionBounds) : CGRectGetMinY(rollingSectionBounds) - headerSize.height,
```

Comment:
What happens if `itemCount == 0` but `showHeaderWhenEmpty` is `NO`?

Reply:
Nothing special, the `headerBounds` will same as `itemCount == 0`.

Before I added this condition, the `headerBounds`'s y position is out off screen when `itemCount == 0`. 

I think this changed did not affect any function because in method 

```objc
- (NSArray<UICollectionViewLayoutAttributes *> *)layoutAttributesForElementsInRect:(CGRect)rect {
        ...
        if (itemCount > 0 || self.showHeaderWhenEmpty) {
            for (NSString *elementKind in _supplementaryAttributesCache.allKeys) {
                NSIndexPath *indexPath = indexPathForSection(section);
                UICollectionViewLayoutAttributes *attributes = [self layoutAttributesForSupplementaryViewOfKind:elementKind
                                                                                                    atIndexPath:indexPath];
                // do not add zero height headers/footers or headers/footers that are outside the rect
                const CGRect frame = attributes.frame;
                const CGRect intersection = CGRectIntersection(frame, rect);
                if (!CGRectIsEmpty(intersection) && CGRectGetLengthInDirection(frame, self.scrollDirection) > 0.0) {
                    [result addObject:attributes];
                }
            }
        }
        ...
}
```

 the header will not be added into the result array.

So, should I modified this into `itemsEmpty && self.showHeaderWhenEmpty`? 😅)

Reply:
Hum... I think I'm wrong, the UI will getting wrong layout. 

I will take a look later.
In my change and set `showHeaderWhenEmpty` to `NO`, try to set a section item count to zero and some error happed.

Reply:
When I trying this, found another bug and note it below.